cmake-modules/AddAzureSDKforC.cmake

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+#
+# How to use: Copy this file to a cmake modules folder within your project.
+# - Add `list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/your-cmake-modules-folder")`
+# - Call this module by doing `include(AddAzureSDKforC)` from the root CMakeLists.txt.
+# - Link your application against your CMake targets, for example, `target_link_libraries(yourApp PRIVATE az_core)`.
+#
+# This will create a _deps folder for the generated project and Azure SDK will be checked out in there.
+# When calling `cmake --build .`, Azure SDK will be built before your application so it can be available for your app to consume.
+#
+# CMake options for Azure SDK for C can be set together with your application CMake options.
+#
+# Note: Update GIT_TAG to the expected version from Azure SDK for C
+#
+
+include(FetchContent)
+FetchContent_Declare(azuresdkforc
+    GIT_REPOSITORY      https://github.com/Azure/azure-sdk-for-c.git
+    GIT_TAG             1.0.0-preview.5)
```

Comment:
Shouldn't this match the "next" version from https://github.com/Azure/azure-sdk-for-c/blob/803d010ece9e442cfc01720f6b13367fdf87c780/sdk/inc/azure/core/az_version.h#L20?

Otherwise, once we ship the next release/tag, it will always be one version behind.